warehouse/views.py

```diff
@@ -192,7 +193,11 @@ def search(request):
     if request.params.getall("c"):
         query = query.filter("terms", classifiers=request.params.getall("c"))
 
-    page_num = int(request.params.get("page", 1))
+    try:
+        page_num = int(request.params.get("page", 1))
+    except ValueError:
+        raise HTTPBadRequest
```

Comment:
You don't want to just use a format like that though, possibly opens you up
to XSS.

On Mon, Jun 6, 2016 at 9:03 PM, Donald Stufft <notifications@github.com>
wrote:

> In warehouse/views.py
> <https://github.com/pypa/warehouse/pull/1242#discussion_r65995809>:
>
> > @@ -192,7 +193,11 @@ def search(request):
> >      if request.params.getall("c"):
> >          query = query.filter("terms", classifiers=request.params.getall("c"))
> >
> > -    page_num = int(request.params.get("page", 1))
> > +    try:
> > +        page_num = int(request.params.get("page", 1))
> > +    except ValueError:
> > +        raise HTTPBadRequest
>
> It'd probably be nice to add a message to this, like raise
> HTTPBadRequest("'page' must be an integer, was
> {!r}".format(request.params["page"]) or something like that. I'm pretty
> sure Pyramid will expose that information in the response so people can see
> what they did wrong.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pypa/warehouse/pull/1242/files/163c42f9716be0173deb7b22968495f5ce79ad57#r65995809>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAADBIOwuSVf6TX1nC4S40FzUFIYFBY9ks5qJMN4gaJpZM4Ivb7P>
> .
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: D1B3 ADC0 E023 8CA6